### DISCUSSION TITLE
Getting search box to include blog posts

### DISCUSSION THREAD

**Discussion by cameronelliott** [thuliteio/doks#395] [Q&A]
Is there some way to get the search-box to include blog articles? 
As far as I can tell, the search box doesn't seem to be including the blog articles.

Thanks,
Cameron

**Comment by h-enk** [comment]
Hey @cameronelliott, nice one!

1. In `./assets/js/index.js`, replace `{{ $list := (where .Site.Pages "Section" "docs") -}}` (line 98) w/: 
    
    ```js
    {{ $list := where site.RegularPages "Type" "in" site.Params.mainSections -}}
    ```
2. Add to `./config/_default/params.toml`:

    ```toml
    # Search
    mainSections = ["blog", "docs"]
    ```

That's it:

![Snag_206caab](https://user-images.githubusercontent.com/3902872/126660268-187746a2-f87c-43db-8f35-d17d3fa7d33b.png)

**Reply by cameronelliott** [reply]
Thanks, @h-enk, that's cool!

**Reply by alistaircom** [reply]
I cannot seem to get this to work. I have tried replacing the line mentioned in `./assets/js/index.js` and adding the line in `./config/_default/params.toml`, but I cannot get blog entries to show up in the search.

So that I understand what this code is meant to do (and perhaps have a better chance at troubleshooting the issue), does `mainSections` refer to the location of the content within the directory structure?  So: 
```
# Search
mainSections = ["blog", "docs"]
```
refers to all documents in: `./content/blog` and `./content/docs`?

**Reply by h-enk** [reply]
>So that I understand what this code is meant to do (and perhaps have a better chance at troubleshooting the issue), does mainSections refer to the location of the content within the directory structure?

Yes, that's correct.

**Reply by alistaircom** [reply]
Ok thanks for that.  I have installed a fresh starter theme, made changes to the two files, added a couple of blog entries, restarted the dev server and I cannot get anything from the blog entries to show in the search results. 

New document pages that are added show up in the search results.

Rather than continue here, would you like me to create an issue for this?

**Reply by h-enk** [reply]
Please share your repo, so I can check what's missing

**Reply by alistaircom** [reply]
https://github.com/alistaircom/my-doks

**Reply by h-enk** [reply]
@alistaircom, thanks for sharing the repo link — it helped!

It turns out, it is a Hugo thing. This is how you make it work:


1. In `./assets/js/index.js`, replace `{{ $list := (where .Site.Pages "Section" "docs") -}}` (line 98) w/: 
    
    ```js
    {{ $list := where site.RegularPages "Type" "in" site.Params.mainSections -}}
    ```
2. Add to `./config/_default/config.toml` (so, not to `./config/_default/params.toml`):

    ```toml
    # Search
    [params]
    mainSections = ['blog', 'docs']
    ```

And there you go:

![Snag_165c589](https://user-images.githubusercontent.com/3902872/129358617-d68dd906-b255-495d-878e-dbee671ab173.png)

**Comment by drwilkins** [comment]
This worked for me. But just FYI for those using the child theme, you'll need to create the folder assets/js/ in the root directory if it doesn't exist and copy the index.js file from node_modules/@hyas/assets/js/index.js to then make the modifications suggested above.

**Reply by roktas** [reply]
Hey @h-enk,

First of all thank you very much for the Doks theme, much appreciated.

Could this `site.Params.mainSections` thingy be a standard logic, to avoid patching `assets/js/index.js` and other files (the ones seen in https://github.com/h-enk/doks/discussions/543, especially https://github.com/h-enk/doks/discussions/543#discussioncomment-1721202)?  I'm hesitating to modify those files while using the child theme.

**Reply by h-enk** [reply]
Hey, nice to hear that you appreciate Doks!

The screenshot in this thread is from an older version of Doks. The latest version of Doks includes the search in the (docs) section  navigation (see also [Better UX navigation](https://getdoks.org/blog/doks-v0.3/#better-ux-navigation)). So, now it wouldn't make sense anymore to combine two sections in the search.

Maybe it would be a something to have an option to have a  site wide search, or a a docs search? I will be giving this some more thought.

**Reply by nk9** [reply]
As someone who is hoping to use the excellent base of Doks on a non-documentation site, with multiple different "post types", it makes more sense in my case to have a single, site-wide search. So I'd very much like to see this supported. I don't mind doing some extra setup to allow this non-standard use.

**Reply by h-enk** [reply]
Added https://github.com/h-enk/doks/issues/630

**Reply by johnnymatthews** [reply]
For folks doing this with the node_module thingy, here's the command you need to run to copy the correct file to the correct place:

```shell
mkdir -p assets/js && cp node_modules/@hyas/doks/assets/js/index.js assets/js
```

**Comment by StefanSalewski** [comment]
Configurable or system wide search would make a lot of sense for v1.0. My current configuration has no docs section at all, but sections books and faq, and a few more, all created with "npm run create -- --kind docs [section]". Well, I have a tiny blog currently, which I may remove later again. For the blog the search may work, but I have deactivated that search box now. For faq it would make sense when it would be working.

**Comment by h-enk** [comment]
Thanks! You've found a bug that ~will be~ is fixed in Doks 1.0.

The files in `./archetypes/docs/*` should have their own `type` (just omit it).  Meaning, your `./archetypes/docs/lorem/ipsum/index.md`  e.g. should look like: 

```md
---
title: "{{ replace .Name "-" " " | title }}"
description: ""
lead: ""
date: {{ .Date }}
lastmod: {{ .Date }}
draft: false
images: []
menu:
  {{ .Section }}:
    parent: "lorem"
    identifier: "{{ .Name }}-{{ delimit (shuffle (split (md5 .Name) "" )) "" }}"
weight: 100
toc: true
---
```

Which will render into:

```md
---
title: "Ipsum"
description: ""
lead: ""
date: 2023-04-15T13:43:59+02:00
lastmod: 2023-04-15T13:43:59+02:00
draft: false
images: []
menu:
  guides:
    parent: "lorem"
    identifier: "ipsum-413b8467abfc839d649b59e84ebbd377"
weight: 100
toc: true
---
```

Now, you can set the sections (types) to include in search — in Doks 1.0 like so (in `./config/_default/hyas/doks.toml`):

```toml
# FlexSearch
flexSearch = true # true (default) or false
indexPages = ["books", "faq"] # [] (all regular pages, default) or list of sections (e.g. ["blog", "guides"])
showSearch = [] # [] (all pages, default) or homepage (optionally) and list of sections (e.g. ["homepage", "blog", "guides"])
```

In Doks 0.5 like so (in `./config/_default/params.toml`):

```toml
[..]
[options]
  [..]
  flexSearch = true
  searchSectionsShow = []
  searchSectionsIndex = ["books", "faq"]
[..]
```

Note that, now (in Doks 0.5) you'll have to copy `./layouts/docs/*` and paste as `./layouts/books/*` and `./layouts/faqs/*`.

Doks 1.0 will use the `./_default/single.html` and `./_default/list.html` layouts — which will then be the layouts for the `docs` based types (like `docs` itself and e.g. `books` and `faqs` (so, no need to set `layout` in the frontmatter).
